MapboxNavigation/Style.swift

```diff
@@ -0,0 +1,311 @@
+    
+    // Table view (Drawer)
+    fileprivate class var defaultHeaderBackground: UIColor { get { return #colorLiteral(red: 1, green: 1, blue: 1, alpha: 1) } }
+    fileprivate class var defaultHeaderTitleLabel: UIColor { get { return defaultPrimaryText } }
+    fileprivate class var defaultHeaderSubtitleLabel: UIColor { get { return defaultSecondaryText } }
+}
+
+fileprivate extension UIFont {
+    // General styling
+    fileprivate class var defaultPrimaryText: UIFont { get { return UIFont.systemFont(ofSize: 16) } }
+    fileprivate class var defaultSecondaryText: UIFont { get { return UIFont.systemFont(ofSize: 16) } }
+    fileprivate class var defaultCellTitleLabel: UIFont { get { return UIFont.systemFont(ofSize: 28, weight: UIFontWeightMedium) } }
+    
+    // Table view (drawer)
+    fileprivate class var defaultHeaderTitleLabel: UIFont { get { return UIFont.preferredFont(forTextStyle: .headline) } }
+}
+
+/**
+ `Style` is a convenient wrapper for styling the appearance of various interface
+ components throughout the Navigation UI.
+ */
+@objc(MBStyle)
+public class Style: NSObject {
+    
+    public var traitCollection: UITraitCollection
+    
+    /**
+     Initializes a style that will be applied for any system traits of an
+     interface’s environment.
+     */
+    convenience override public init() {
+        self.init(traitCollection: UITraitCollection())
+    }
+    
+    /**
+     Initializes a style for a specific system trait(s) of an interface’s
+     environment.
+     */
+    required public init(traitCollection: UITraitCollection) {
+        self.traitCollection = traitCollection
+        super.init()
+    }
+    
+    ///  General styling
+    
+    /**
+     `tintColor` is used for guidance arrow, highlighted text, progress bar and
+     more.
+     */
+    public var tintColor: UIColor?
+    
+    /**
+     `primaryTextColor` sets the color for titles and other prominent information.
+     */
+    public var primaryTextColor: UIColor?
+    
+    /**
+     `secondaryTextColor` sets the color for subtitles and other subtle
+     information.
+    */
+    public var secondaryTextColor: UIColor?
+    
+    /**
+     `buttonTextColor` sets the text color on buttons for normal state.
+     */
+    public var buttonTextColor: UIColor?
+    
+    /**
+     `lineColor` sets the color of dividers and separators.
+     */
+    public var lineColor: UIColor?
+    
+    /**
+     `routeCasingColor` sets the casing color of the route line.
+     
+     Note that this color will not be used if you are implementing
+     `routeCasingStyleLayerWithIdentifier(_:identifier:source:)`.
+     */
+    public var routeCasingColor: UIColor?
+    
+    /**
+     `routeLayerColor` sets the color of the route line.
+     
+     Note that this color will not be used if you are implementing
+     `routeCasingStyleLayerWithIdentifier(_:identifier:source:)`.
+     */
+    public var routeLayerColor: UIColor?
```

Comment:
I think we’re in agreement: https://github.com/mapbox/mapbox-navigation-ios/pull/162#discussion_r112728486.